datasafe-business-impl/src/test/java/de/adorsys/datasafe/business/impl/cmsencryption/CmsEncryptionServiceImplTest.java

```diff
@@ -32,37 +31,144 @@
+        ByteArrayInputStream inputStream = new ByteArrayInputStream(byteArray);
+        InputStream decryptionStream = cmsEncryptionService.buildDecryptionInputStream(inputStream, keyStoreAccess);
+        byte[] actualResult = IOUtils.toByteArray(decryptionStream);
 
-		assertThat(MESSAGE_CONTENT).isEqualTo(new String(actualResult));
-		log.debug("en and decrypted successfully");
-	}
+        assertThat(MESSAGE_CONTENT).isEqualTo(new String(actualResult));
+        log.debug("en and decrypted successfully");
+    }
+
+    @Test
+    @SneakyThrows
+    @Ignore
+    public void cmsEnvelopeEncryptAndDecryptFileStreamTest() {
+        String folderPath = "target/";
+        String testFilePath = folderPath + "test.dat";
+        String encryptedFilePath = folderPath + "test_encrypted.dat";
+        String decryptedTestFilePath = folderPath + "test_decrypted.dat";
+
+        int _1MbInBytes = 1024 / 1024;
+        int testFileSizeInBytes = 1024 * 1024 * 1024; //1Gb
+        double freeSpaceThresholdCoeff = 3.1;
+
+        log.info("For the test, needed {}Mb free space", 3.1 * testFileSizeInBytes / _1MbInBytes);
+        File testFilesDirectory = new File("target");
+        if(testFilesDirectory.getFreeSpace() < testFileSizeInBytes * freeSpaceThresholdCoeff) {
+            log.debug("Free disk space: {}, Size of one test file: {}", testFilesDirectory.getFreeSpace(), testFileSizeInBytes);
+            Assertions.fail("Free space on the disk isn't enough for test, encrypted and decrypted files");
+        }
+        generateTestFile(testFilePath, testFileSizeInBytes);
+        log.info("Test file with size {}Mb generated: {}", testFileSizeInBytes / _1MbInBytes, testFilePath);
+
+
+        ReadKeyPassword readKeyPassword = new ReadKeyPassword("readkeypassword");
+        ReadStorePassword readStorePassword = new ReadStorePassword("readstorepassword");
+        KeyStoreAuth keyStoreAuth = new KeyStoreAuth(readStorePassword, readKeyPassword);
+
+        KeyStoreCreationConfig config = new KeyStoreCreationConfig(1, 0, 1);
+
+        KeyStore keyStore = keyStoreService.createKeyStore(keyStoreAuth, KeyStoreType.DEFAULT, config);
+        KeyStoreAccess keyStoreAccess = new KeyStoreAccess(keyStore, keyStoreAuth);
+        PublicKeyIDWithPublicKey publicKeyIDWithPublicKey = keyStoreService.getPublicKeys(keyStoreAccess).get(0);
+
+        File testFile = new File(testFilePath);
+        FileInputStream fisTestFile = new FileInputStream(testFile);
+
+        File encryptedFile = new File(encryptedFilePath);
+        FileOutputStream fosEnFile = new FileOutputStream(encryptedFile);
+        log.debug("Total file size to read (in Mb) : " + fisTestFile.available() / _1MbInBytes);
+
+        OutputStream encryptionStream = cmsEncryptionService.buildEncryptionOutputStream(
+                fosEnFile, publicKeyIDWithPublicKey.getPublicKey(), publicKeyIDWithPublicKey.getKeyID());
+        transferFromInputToOutputStream(fisTestFile, encryptionStream);
```

Comment:
Can't we pipe data from some data generator instead of dealing with file? This generator could calculate MD5 hash as well, it would really simplify this test